### DISCUSSION TITLE
Can you exclude directories?

### DISCUSSION THREAD

**Discussion by musicpanda** [WinMerge/winmerge#1571] [General]
Among other things I use Winmerge to maintain webshops.

Comparing the software of a problematic shop (with bugs) with a standard shop tells me quickly where the software has been changed and that helps to find the causes of the instability. 

The first directory level consists of about a dozen of directories, All are of interest - except for the one that contains product images. Unfortunately that image directory can become several gigabytes big so it would be nice if there was some way to exclude it.

**Comment by sdottaka** [comment]
In recent versions, the `res` folder will be ignored if you specify `!res\` in the Select files or folders window, as shown below. Or you can define a file filter.

![image](https://user-images.githubusercontent.com/98126/201701547-918af125-a404-4ceb-a3c4-59c40f4d71bd.png)
![image](https://user-images.githubusercontent.com/98126/201702032-dffef1da-3edd-4cf8-8cc1-0ae6b69dfd6f.png)

**Reply by PhantomLord72** [reply]
This answer is worth more than 100 pages of manual. Thank you @sdottaka

**Comment by musicpanda** [comment]
Thank you!

**Comment by musicpanda** [comment]
Just to be complete: does this method also allow to skip more than one directory?

**Comment by sdottaka** [comment]
To ignore multiple folders, specify them separated by semicolons like `!res\;!img\`

**Comment by kingsqueens** [comment]
Hi there, 
how to exclude more than one directory using the filter (flt file setting) ?
I aim to create a filter with regular expressions for different folders and files and now to start I'm trying to create the equivalent filter to "!res\;!img\" using the filter but without success.

I've tried:
xd: \\abc
but the comparison is not filtering out the folder and I'd expect to not see this folder in the comparison.

anyone could help me ?
thanks,

**Reply by sdottaka** [reply]
If you define and use a filter as shown below, the res and img folders will be excluded.

~~~
name: #1571
desc: #1571 Longer description
def: include
d: \\res$
d: \\img$
~~~
